src/org/centos/pipeline/PackagePipelineUtils.groovy

```diff
@@ -475,3 +475,148 @@ def setScratchVars(Map parsedMsg) {
+        sh  script: './ghi comment ' + env.ghprbPullId + ' -m "' + msg + '"', label: "Commenting"
+    }
+}
+
+/**
+    * info about tags to be used
+    * @param map
+    */
+def printLabelMap(map) {
+    for (tag in map) {
+        echo "tag to be used for ${tag.key} -> ${tag.value}"
+    }
+}
+
+/**
+    * Setup container templates in openshift
+    * @param openshiftProject Openshift Project
+    * @return
+    */
+def setupContainerTemplates(String openshiftProject) {
+    openshift.withCluster() {
+        openshift.withProject(openshiftProject) {
+            dir('config/s2i') {
+                sh script: './create-containers.sh', label: "Executing create-containers.sh script"
+            }
+        }
+    }
+}
+
+/**
+    * Build image in openshift
+    * @param openshiftProject Openshift Project
+    * @param buildConfig
+    * @return
+    */
+def buildImage(String openshiftProject, String buildConfig) {
+    // - build in Openshift
+    // - startBuild with a commit
+    // - Get result Build and get imagestream manifest
+    // - Use that to create a unique tag
+    // - This tag will then be passed as an image input
+    //   to the podTemplate/containerTemplate to create
+    //   our slave pod.
+    openshift.withCluster() {
+        openshift.withProject(openshiftProject) {
+            def result = openshift.startBuild(buildConfig,
+                    "--commit",
+                    "refs/pull/" + env.ghprbPullId + "/head",
+                    "--wait")
+            def out = result.out.trim()
+            echo "Resulting Build: " + out
+
+            def describeStr = openshift.selector(out).describe()
+            outTrim = describeStr.out.trim()
+
+            // --wait is being lost due to socket timeouts
+            buildRunning = true
+            while (buildRunning) {
+                describeStr = openshift.selector(out).describe()
+                outTrim = describeStr.out.trim()
+                buildRunning = sh(script: "echo \"${outTrim}\" | grep '^Status:' | grep -E 'New|Pending|Running'", returnStatus: true) == 0
+                sleep 60
+            }
+
+            def imageHash = sh(
+                    script: "echo \"${outTrim}\" | grep 'Image Digest:' | cut -f2- -d:",
+                    returnStdout: true
+            ).trim()
+            echo "imageHash: ${imageHash}"
+
+            echo "Creating CI tag for ${openshiftProject}/${buildConfig}: ${buildConfig}:PR-${env.ghprbPullId}"
+
+            openshift.tag("${openshiftProject}/${buildConfig}@${imageHash}",
+                    "${openshiftProject}/${buildConfig}:PR-${env.ghprbPullId}")
+
+            return "PR-" + env.ghprbPullId
+        }
+    }
+}
+
+/**
+    * Build stable image in openshift
+    * @param openshiftProject Openshift Project
+    * @param buildConfig
+    * @return
+    */
+def buildStableImage(String openshiftProject, String buildConfig, String buildTag = 'stable') {
```

Comment:
Same thing as buildImage, it is just a different tag. I would not add this here and have more things to maintain, and just use https://github.com/openshift/contra-lib/blob/master/vars/buildOpenShiftImage.groovy